### PR TITLE
Improve CupertinoDatePicker docs

### LINKED ISSUE #57208: [CupertinoDatePicker] specify hours available for picking (making other hours disabled)
Hi,
I have used cupertinoDatePicker mode time inside  alertDialog.I want The time should be a list as shown. With the times that are booked needs to be greyed out.like as given below please given any
reference.
 
image

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
## Description

Docs only change. 

## Related Issues

Fixes https://github.com/flutter/flutter/issues/57208

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

**Review by justinmc — APPROVED** [review]
LGTM, seems much more specific and clear!

File: packages/flutter/lib/src/cupertino/date_picker.dart [hunk 1]
```diff
@@ -211,15 +211,15 @@ class CupertinoDatePicker extends StatefulWidget {
   /// the intervals set in [minimumDate], [maximumDate], [minimumYear], and
   /// [maximumYear].
   ///
-  /// [minimumDate] is the minimum [DateTime] that the picker can be scrolled to.
-  /// Null if there's no limit. In [CupertinoDatePickerMode.time] mode, if the
-  /// date part of [initialDateTime] is after that of the [minimumDate], [minimumDate]
-  /// has no effect.
+  /// [minimumDate] is the minimum selectable [DateTime] of the picker. In
+  /// [CupertinoDatePickerMode.time] mode, [minimumDate] should typically be on
+  /// the same date as [initialDateTime]. When set to null, the picker does not
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/date_picker.dart:216]: Is this right? it's a bit different from the original doc. What happens if it's not?

**Review by xster — COMMENTED** [review]
Thanks for the cleanup

File: packages/flutter/lib/src/cupertino/date_picker.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/cupertino/date_picker.dart]: Updated the doc to explain the suggestion. I believe they're roughly equivalent (has no effect if too early, invalid if too late)?

File: packages/flutter/lib/src/cupertino/date_picker.dart [hunk 2]
```diff
@@ -211,15 +211,17 @@ class CupertinoDatePicker extends StatefulWidget {
...
   ///
-  /// [maximumDate] is the maximum [DateTime] that the picker can be scrolled to.
-  /// Null if there's no limit. In [CupertinoDatePickerMode.time] mode, if the
-  /// date part of [initialDateTime] is before that of the [maximumDate], [maximumDate]
-  /// has no effect.
+  /// [maximumDate] is the maximum selectable [DateTime] of the picker. When set
+  /// to null, the picker does not limit the maximum [DateTime] the user can pick.
+  /// In [CupertinoDatePickerMode.time] mode, [maximumDate] should typically be
+  /// on the same date as [initialDateTime], as the picker will not limit the
+  /// maxium time the user can pick if it's set to a date later than that.
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/date_picker.dart:224]: typo

**Review by xster — APPROVED** [review]

**Pull request merged**